ir/instr.cpp

```diff
@@ -836,86 +853,139 @@ static StateValue fm_poison(State &s, expr a, const expr &ap,
                    ty, fmath, rm, bitwise, flags_in_only, 1);
 }
 
+static StateValue uf_float(const string &name,
+                           const vector<StateValue> &args,
+                           const expr &res,
+                           FastMathFlags fmath = FastMathFlags(),
+                           bool is_commutative = false) {
+  
+  vector<expr> arg_values;
+  arg_values.reserve(args.size());
+  for (auto &arg : args) {
+    arg_values.push_back(arg.value);
+  }
+
+  auto value = expr::mkUF(name, arg_values, res);
+  if (is_commutative) {
+    assert(args.size() == 2);
+    value = value & expr::mkUF(name, {arg_values[1], arg_values[0]}, res);
+  }
+
+  AndExpr non_poison;
+  for (auto &arg : args) {
+    non_poison.add(arg.non_poison);
+  }
+
+  auto fast_math_flag = [&](unsigned flag, const char* suffix){
+    if (fmath.flags & flag) {
+      auto np_name = name + ".np_" + suffix;
+      auto poison_uf = expr::mkUF(np_name, arg_values, false);
+      if (is_commutative) {
+        assert(args.size() == 2);
+        poison_uf &= expr::mkUF(np_name, {arg_values[1], arg_values[0]}, false);
+      }
+      non_poison.add(poison_uf);
+    }
+  };
+  
+  fast_math_flag(FastMathFlags::NNaN, "nnan");
```

Comment:
I don't understand this part. Also, where are the other flags?

Reply:
It handles flags which may result in the operation returning poison. As far as I can tell the only flags which can result in poison are `nnan` and `ninf`. They are handled by creating additional uninterpreted functions `op.np_nnan(x, y): bool` and `op.np_ninf(x, y): bool` (in the binary case) whose results are added to the `non_poison` and expression.

The `nsz` flag is not relevant as in uf-float mode we do not really have a zero value.

I don't think I entirely understand how the remaining fast math flags are handled in the existing code, though it looks like the result of the operation is wrapped in a function call to another UF.